### PR TITLE
Add API endpoints for getting action jobs status

### PR DESCRIPTION
**Pull request by chesteripz**
Sample of response, it is similar to Github actions

ref https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository

``` json
{
    "workflow_runs": [
        {
            "id": 3,
            "name": "Explore-Gitea-Actions",
            "head_branch": "main",
            "head_sha": "6d8d29a9f7a01ded8f8aeb64341cb31ee1ab5f19",
            "run_number": 3,
            "event": "push",
            "display_title": "More job",
            "status": "success",
            "workflow_id": "demo2.yaml",
            "url": "/chester/test/actions/runs/3",
            "created_at": "2023-08-22T13:41:33-04:00",
            "updated_at": "2023-08-22T13:41:37-04:00",
            "run_started_at": "2023-08-22T13:41:33-04:00"
        },
        {
            "id": 2,
            "name": "Explore-Gitea-Actions",
            "head_branch": "main",
            "head_sha": "6d8d29a9f7a01ded8f8aeb64341cb31ee1ab5f19",
            "run_number": 2,
            "event": "push",
            "display_title": "More job",
            "status": "success",
            "workflow_id": "demo.yaml",
            "url": "/chester/test/actions/runs/2",
            "created_at": "2023-08-22T13:41:30-04:00",
            "updated_at": "2023-08-22T13:41:33-04:00",
            "run_started_at": "2023-08-22T13:41:30-04:00"
        },
        {
            "id": 1,
            "name": "Explore-Gitea-Actions",
            "head_branch": "main",
            "head_sha": "e5369ab054cae79899ba36e45ee82811a6e0acd5",
            "run_number": 1,
            "event": "push",
            "display_title": "Add job",
            "status": "failure",
            "workflow_id": "demo.yaml",
            "url": "/chester/test/actions/runs/1",
            "created_at": "2023-08-22T13:15:21-04:00",
            "updated_at": "2023-08-22T13:18:10-04:00",
            "run_started_at": "2023-08-22T13:15:21-04:00"
        }
    ],
    "total_count": 3
}
```

### CODE REVIEW

File: routers/api/v1/api.go
Comment (techknowlogick) [inline routers/api/v1/api.go:1172]: Thank you very much for your contribution 🙏 To align with GitHub Actions so that other tools can read similar data from the API endpoints, can you adjust the URL to match the format `/repos/{owner}/{repo}/actions/runs` and the output schema to be something like (maybe not all the fields are needed yet):
```
{
  "total_count": 1,
  "workflow_runs": [
    {
      "id": 30433642,
      "name": "Build",
      "head_branch": "master",
      "head_sha": "acb5820ced9479c074f688cc328bf03f341a511d",
      "run_number": 562,
      "event": "push",
      "display_title": "Update README.md",
      "status": "queued",
      "conclusion": null,
      "workflow_id": 159038,
      "created_at": "2020-01-22T19:33:08Z",
      "updated_at": "2020-01-22T19:33:08Z",
      "actor": {
// ...
      },
      "run_started_at": "2020-01-22T19:33:08Z",
      "triggering_actor": {
     // ...
      },
     
    }
  ]
}
```

If you have any questions or aren't able to take on the above due to time constraints or other reason, please let me know and I can submit the changes.

File: routers/api/v1/api.go
Comment (yp05327) [inline routers/api/v1/api.go]:
```suggestion
				}, reqRepoReader(unit.TypeActions), context.ReferencesGitRepo(true))
```

**Comment by puni9869** [comment]
@chesteripz  There are still some linter issues.

**Comment by chesteripz** [comment]
Should resolve #25572

File: routers/api/v1/api.go
Comment (lunny) [reply routers/api/v1/api.go:1172]: @chesteripz please resolve the change request.

File: modules/structs/repo_actions.go
Comment (puni9869) [inline modules/structs/repo_actions.go]:
```suggestion
	CreatedAt time.Time `json:"created_at"`
```

File: modules/structs/repo_actions.go
Comment (puni9869) [inline modules/structs/repo_actions.go]:
```suggestion
	UpdatedAt time.Time `json:"updated_at"`
```

File: modules/structs/repo_actions.go
Comment (puni9869) [inline modules/structs/repo_actions.go]:
```suggestion
	RunStartedAt time.Time `json:"run_started_at"`
```

Comment (puni9869) [reply modules/structs/repo_actions.go]: If there is change in swagger schema , please run `make generate-swagger`

File: routers/api/v1/repo/actions.go
Comment (puni9869) [inline routers/api/v1/repo/actions.go:45]: If applicable, could you include these in swagger

```
//    "400":
	//   "403":
	//   "404":
	//   "409":
	//   "422":
```

**Comment by chesteripz** [comment]
> Hi, any plans of getting this merged and releasing it? Would love to help if any is required

I actually dont know what else is needed for this to be merged

**Comment by lunny** [comment]
Please fix the CI.

**Comment by chesteripz** [comment]
> Please fix the CI.

It's security check failing. https://github.com/go-gitea/gitea/actions/runs/8547937533/job/23420914824?pr=26673

To my best knowledge, this requires upgrading to 1.22.2+ to resolve. https://pkg.go.dev/vuln/GO-2024-2687

And confirmed the same error on main branch

```
=== Informational ===

There is 1 vulnerability in modules that you require that is neither
imported nor called. You may not need to take any action.
See https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck for details.

Vulnerability #1: GO-2022-0470
    No access control in github.com/blevesearch/bleve and bleve/v2
  More info: https://pkg.go.dev/vuln/GO-2022-0470
  Module: github.com/blevesearch/bleve/v2
    Found in: github.com/blevesearch/bleve/v2@v2.3.10
    Fixed in: N/A

Your code is affected by 1 vulnerability from the Go standard library.

Share feedback at https://go.dev/s/govulncheck-feedback.
exit status 3
make: *** [Makefile:786: security-check] Error 1

❯ git rev-parse HEAD
663acd0b4620a7a4e83f2d0699749af95126b0f5
```

**Comment by chesteripz** [comment]
If anyone want to make changes to this PR, feel free to do so.

**Comment by chesteripz** [comment]
https://github.com/actions/go-versions/pull/98/files

Until this gets merged, `actions/setup-go@v5` will be stuck with go 1.22.1

File: routers/api/v1/repo/actions.go
Comment (yp05327) [inline routers/api/v1/repo/actions.go]: this can be `utils.GetListOptions(ctx)`

File: modules/structs/repo_actions.go
Comment (yp05327) [inline modules/structs/repo_actions.go:27]: Maybe `started` and `duration` ( ActionTask.Duration() ) is better?

File: services/convert/convert.go
Comment (yp05327) [inline services/convert/convert.go]: maybe we need return error or empty object here, or we may get panic bellow?

File: modules/structs/repo_actions.go
Comment (chesteripz) [reply modules/structs/repo_actions.go:27]: the initial PR contains duration attribute, but as @techknowlogick suggested, it's probably better to align with Github action endpoints, the discussion: https://github.com/go-gitea/gitea/pull/26673#pullrequestreview-1590585963

File: routers/api/v1/repo/actions.go
Comment (lunny) [inline routers/api/v1/repo/actions.go]: Repository ID need to become the options field which is missed.

File: services/convert/convert.go
Comment (lunny) [inline services/convert/convert.go]: All Link suffix will only return url without host and port. It's not right for an API result?

**Review by lunny — CHANGES_REQUESTED** [review]
Some small nits needs to be handled.

File: services/convert/convert.go
Comment (lunny) [inline services/convert/convert.go]: Could we return error but not panic?

Comment (chesteripz) [reply services/convert/convert.go]: do we prefer empty list response on error? or 500 in that case?

Comment (lunny) [reply services/convert/convert.go]: I think we should return 500 and with no error detail but log the error.

File: routers/api/v1/repo/actions.go
Comment (lunny) [inline routers/api/v1/repo/actions.go:55]: 500 needs to be document.

Comment (chesteripz) [reply routers/api/v1/repo/actions.go:55]: Do we document `500` anywhere else?

Comment (lunny) [reply routers/api/v1/repo/actions.go:55]: I think not. I'm also weird for that.

**Review by lunny — APPROVED** [review]

File: routers/api/v1/repo/actions.go
Comment (yp05327) [inline routers/api/v1/repo/actions.go]: 409 is conflict

File: modules/structs/repo_actions.go
Comment (yp05327) [inline modules/structs/repo_actions.go]:
```suggestion
	HeadSHA      string `json:"head_sha"`
```

File: services/convert/convert.go
Comment (yp05327) [inline services/convert/convert.go]:
```suggestion
func ToActionTask(ctx context.Context, t *actions_model.ActionTask) (*api.ActionTask, error) {
```

**Comment by yp05327** [comment]
Others LGTM, I will approve after the fix.

**Review by yp05327 — APPROVED** [review]

**Pull request merged**
